### PR TITLE
add goal to run tests in makefile

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,13 @@ generate: vendor
 	--bounding-dirs github.com/codeready-toolchain/api/pkg/apis "toolchain:v1alpha1" \
 	--go-header-file=make/go-header.txt
 	
-	
+.PHONY: test
+## runs the tests
+test:
+	@echo "running the tests..."
+	$(Q)GO111MODULE=on \
```
Comment (dipak-pawar) [inline Makefile]: can we export `GO111MODULE=on` only once?

Comment (xcoulon) [reply Makefile]: @dipak-pawar I tried to set the variable once but it didn't work so for now I'm doing this. I can revisit this part later if needed

Comment (xcoulon) [reply Makefile]: but if this doesn't block, I'd like to move on so we can have the OpenShift CI in place for this repo

Comment (xcoulon) [reply Makefile]: ah wait, I can do this, indeed:
```
GO111MODULE?=on
export GO111MODULE
```
let me change that

Comment (xcoulon) [reply Makefile]: done in https://github.com/codeready-toolchain/api/pull/8/commits/e23614cb5689a4af4839bf9ed6aa3a16ce0285e8

**Pull request merged**